test/unit/FutarchyChallengeFactory.js

```diff
@@ -0,0 +1,137 @@
+import lkTestHelpers from 'lk-test-helpers'
+const { expectRevert, expectEvent } = lkTestHelpers(web3)
+
+const FutarchyChallengeFactory = artifacts.require('FutarchyChallengeFactory.sol')
+const FutarchyChallenge = artifacts.require('FutarchyChallenge.sol')
+const FutarchyOracleMock = artifacts.require('FutarchyOracleMock.sol')
+const RegistryMock = artifacts.require('RegistryMock.sol')
+const ScalarPriceOracleFactoryMock = artifacts.require('ScalarPriceOracleFactoryMock')
+const DutchExchangeMock = artifacts.require('DutchExchangeMock')
+const Token = artifacts.require('EIP20.sol')
+const BigNumber = require('bignumber.js');
+
+contract('FutarchyChallengeFactory', (accounts) => {
+  let futarchyChallengeFactory
+
+  const mockAddresses = [
+    '0x4e0100882b427b3be1191c5a7c7e79171b8a24dd',
+    '0x6512df5964f1578a8164ce93a3238f2b11485d1c',
+    '0x687355ca7a320e5420a3db5ae59ef662e4146786'
+  ]
+
+  let COMPARATOR_TOKEN, SCALAR_PRICE_ORACLE_FACTORY, DUTCH_EXCHANGE // async "const's"
```

Comment:
subpar solution :/

Reply:
Why not set these here instead of deployChallengeFactory if they're only deployed once anyway?

Reply:
Doing async transactions at the `describe` or `contract` level seems to make things wonky, including wonky test logging and you can't call `only` on any test clauses.